File: goopylib/src/extension/color/random.cpp
#include "random.h"

namespace color {
    PyObject *random_rgb(PyObject *Py_UNUSED(self)) {
        gp::ColorRGB rgb = gp::randomRGB();
        return color::createRGB(rgb.getRed(), rgb.getGreen(), rgb.getBlue());
    }

    PyObject *random_hex(PyObject *Py_UNUSED(self)) {
        gp::ColorHex hex = gp::randomHex();
        return color::createHex(hex.toString().c_str());
    }

    PyObject *random_cmyk(PyObject *Py_UNUSED(self)) {
        gp::ColorCMYK cmyk = gp::randomCMYK();
        return color::createCMYK(cmyk.getCyan(), cmyk.getMagenta(), cmyk.getYellow(), cmyk.getKey());
    }

    PyObject *random_hsv(PyObject *Py_UNUSED(self)) {
        gp::ColorHSV hsv = gp::randomHSV();
        return color::createHSV(hsv.getHue(), hsv.getSaturation(), hsv.getValue());
    }

    PyObject *random_hsl(PyObject *Py_UNUSED(self)) {
        gp::ColorHSL hsl = gp::randomHSL();
        return color::createHSV(hsl.getHue(), hsl.getSaturation(), hsl.getLuminance());
    }
}

File: goopylib/color/random.py
"""
Use this module for simple random color generation in different formats.
"""

from __future__ import annotations

from random import randint as _randint
from random import random as _random

# pylint: disable-next=no-name-in-module
from . import ColorRGB, ColorHex, ColorCMYK, ColorHSV, ColorHSL


# Random Colors

def random_rgb() -> ColorRGB:
    """
    Returns:
         a random RGB color
    """
    return ColorRGB(_randint(0, 255), _randint(0, 255), _randint(0, 255))


def random_hex() -> ColorHex:
    """
    Returns:
         a random Hexadecimal color
    """
    return ColorHex(random_rgb())


def random_cmyk() -> ColorCMYK:
    """
    Returns:
         a random CMYK color
    """
    return ColorCMYK(_random(), _random(), _random(), _random())


def random_hsv() -> ColorHSV:
    """
    Returns:
        a random HSV color
    """
    return ColorHSV(_randint(0, 360), _random(), _random())


def random_hsl() -> ColorHSL:
    """
    Returns:
         a random HSL color
    """
    return ColorHSL(_randint(0, 360), _random(), _random())
